packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -172,8 +173,16 @@ export const createAddressTransactionsProvider = ({
                   // Rollback by 1 block, try again in next loop iteration
                   localTransactions = localTransactions.filter(({ blockHeader: { blockNo } }) => blockNo < lowerBound);
                 } else {
-                  localTransactions = unionBy(localTransactions, newTransactions, (tx) => tx.id);
-                  store.setAll(localTransactions);
+                  const lastLocalTxs = localTransactions.slice(-newTransactions.length);
+                  const areTransactionsSame =
+                    lastLocalTxs.length === newTransactions.length &&
+                    lastLocalTxs.every((tx, index) => tx.id === newTransactions[index].id);
```

Comment:
This comparison could be skipped entirely by checking the length of the already computed `duplicateTransactions`.
If it has the same length as newTransactions, it means that all new transactions are already part of the localTransactions.
In this case, it's not even necessary to compare the length anymore.
```diff
- lastLocalTxs.length === newTransactions.length &&
- lastLocalTxs.every((tx, index) => tx.id === newTransactions[index].id);
+ newTransactions.length === duplicateTransactions.length;
```